### DISCUSSION TITLE
Access to TT core

### DISCUSSION THREAD

**Discussion by nicolacavallini** [ion-g-ion/torchTT#10] [Q&A]
Dears,

I'm looking for functionality that allows me to access the single core of a Tensor Train. 

I found pretty convenient with the TT toolbox the ability to do this:
```
core = tensor_train{dimension_index}
```
or: 
```
tensor_train{dimension_index} = core
```
Do you have an analogous mechanism? Apologies if I missed it in the documentation. 

I can help with the code if necessary. 

Best,
Nicola

**Comment by ion-g-ion** [comment]
Dear Nicola,

The cores can be accessed via the .cores attribute (for example `tt.cores[1]` returns the second core of the tensor). It is not a write safe attribute and only read only access should be done this way. Using the constructor can be used to create a tensor from a list of cores `tt = torchtt.TT(cores)` and the newly created method `set_cores` should be used to update individual cores.

Best regards,
Ion

**Reply by nicolacavallini** [reply]
Beautiful!
